ethportal-api/src/types/content_value/beacon.rs

```diff
@@ -51,23 +67,348 @@ impl<'de> Deserialize<'de> for PossibleBeaconContentValue {
+        };
+
+        Ok(Self {
+            fork_name,
+            update: content,
+        })
+    }
+}
+
+impl Decode for ForkVersionedLightClientOptimisticUpdate {
+    fn is_ssz_fixed_len() -> bool {
+        false
+    }
+
+    fn from_ssz_bytes(bytes: &[u8]) -> Result<Self, DecodeError> {
+        Self::decode(bytes).map_err(|err| DecodeError::BytesInvalid(format!("{err:?}")))
+    }
+}
+
+impl Encode for ForkVersionedLightClientOptimisticUpdate {
+    fn is_ssz_fixed_len() -> bool {
+        false
+    }
+
+    fn ssz_append(&self, buf: &mut Vec<u8>) {
+        buf.extend_from_slice(&self.encode());
+    }
+
+    fn ssz_bytes_len(&self) -> usize {
+        self.encode().len()
+    }
+}
+
+/// A wrapper type including a `ForkName` and `LightClientFinalityUpdate`
+#[derive(Clone, Debug, PartialEq)]
+pub struct ForkVersionedLightClientFinalityUpdate {
+    pub fork_name: ForkName,
+    pub update: LightClientFinalityUpdate,
+}
+
+impl ForkVersionedLightClientFinalityUpdate {
+    fn encode(&self) -> Vec<u8> {
+        let fork_digest = self.fork_name.as_fork_digest();
+
+        let mut data = fork_digest.to_vec();
+        data.extend(self.update.as_ssz_bytes());
+        data
+    }
+
+    fn decode(buf: &[u8]) -> Result<Self, DecodeError> {
+        let fork_digest = ForkDigest::try_from(&buf[0..4]).map_err(|err| {
+            DecodeError::BytesInvalid(format!("Unable to decode fork digest: {err:?}"))
+        })?;
+
+        let fork_name = ForkName::try_from(fork_digest).map_err(|_| {
+            DecodeError::BytesInvalid(format!("Unable to decode fork name: {fork_digest:?}"))
+        })?;
+
+        let content = match fork_name {
+            ForkName::Bellatrix => LightClientFinalityUpdate::Bellatrix(
+                LightClientFinalityUpdateBellatrix::from_ssz_bytes(&buf[4..])?,
+            ),
+            ForkName::Capella => LightClientFinalityUpdate::Capella(
+                LightClientFinalityUpdateCapella::from_ssz_bytes(&buf[4..])?,
+            ),
+        };
+
+        Ok(Self {
+            fork_name,
+            update: content,
+        })
+    }
+}
+
+impl Decode for ForkVersionedLightClientFinalityUpdate {
+    fn is_ssz_fixed_len() -> bool {
+        false
+    }
+
+    fn from_ssz_bytes(bytes: &[u8]) -> Result<Self, DecodeError> {
+        Self::decode(bytes)
+    }
+}
+
+impl Encode for ForkVersionedLightClientFinalityUpdate {
+    fn is_ssz_fixed_len() -> bool {
+        false
+    }
+
+    fn ssz_append(&self, buf: &mut Vec<u8>) {
+        buf.extend_from_slice(&self.encode());
+    }
+
+    fn ssz_bytes_len(&self) -> usize {
+        self.encode().len()
+    }
+}
+
+/// Maximum number of `LightClientUpdate` instances in a single request is 128;
+/// Defined in https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/p2p-interface.md#configuration
+#[derive(Clone, Debug, PartialEq)]
+pub struct LightClientUpdatesByRange(VariableList<ForkVersionedLightClientUpdate, U128>);
```

Comment:
*nit* it would be nice to group up the `ForkVersionedLightClientUpdate` near this definition, instead of having the ...`FinalityUpdate` split them up. A code reader is probably interested in `ForkVersionedLightClientUpdate` and `LightClientUpdatesByRange` at the same time.